tests/dragonfly/cluster_test.py

```diff
@@ -835,5 +835,128 @@ async def test_cluster_slot_migration(df_local_factory: DflyInstanceFactory):
+    )
+
+    await c_nodes_admin[0].close()
+    await c_nodes_admin[1].close()
+
+
+@dfly_args({"proactor_threads": 4, "cluster_mode": "yes"})
+async def test_cluster_data_migration(df_local_factory: DflyInstanceFactory):
+    # Check data migration from one node to another
+    nodes = [
+        df_local_factory.create(port=BASE_PORT + i, admin_port=BASE_PORT + i + 1000)
+        for i in range(2)
+    ]
+
+    df_local_factory.start_all(nodes)
+
+    c_nodes = [node.client() for node in nodes]
+    c_nodes_admin = [node.admin_client() for node in nodes]
+
+    node_ids = await asyncio.gather(*(get_node_id(c) for c in c_nodes_admin))
+
+    config = f"""
+      [
+        {{
+          "slot_ranges": [
+            {{
+              "start": 0,
+              "end": LAST_SLOT_CUTOFF
+            }}
+          ],
+          "master": {{
+            "id": "{node_ids[0]}",
+            "ip": "localhost",
+            "port": {nodes[0].port}
+          }},
+          "replicas": []
+        }},
+        {{
+          "slot_ranges": [
+            {{
+              "start": NEXT_SLOT_CUTOFF,
+              "end": 16383
+            }}
+          ],
+          "master": {{
+            "id": "{node_ids[1]}",
+            "ip": "localhost",
+            "port": {nodes[1].port}
+          }},
+          "replicas": []
+        }}
+      ]
+    """
+
+    await push_config(
+        config.replace("LAST_SLOT_CUTOFF", "9000").replace("NEXT_SLOT_CUTOFF", "9001"),
+        c_nodes_admin,
+    )
+
+    assert await c_nodes[0].set("KEY0", "value")
+    assert await c_nodes[0].set("KEY1", "value")
+    assert await c_nodes[1].set("KEY2", "value")
+    assert await c_nodes[1].set("KEY3", "value")
+    assert await c_nodes[0].set("KEY4", "value")
+    assert await c_nodes[0].set("KEY5", "value")
+    assert await c_nodes[1].set("KEY6", "value")
+    assert await c_nodes[1].set("KEY7", "value")
+    assert await c_nodes[0].set("KEY8", "value")
+    assert await c_nodes[0].set("KEY9", "value")
+    assert await c_nodes[1].set("KEY10", "value")
+    assert await c_nodes[1].set("KEY11", "value")
+    assert await c_nodes[0].set("KEY12", "value")
+    assert await c_nodes[0].set("KEY13", "value")
+    assert await c_nodes[1].set("KEY14", "value")
+    assert await c_nodes[1].set("KEY15", "value")
+    assert await c_nodes[0].set("KEY16", "value")
+    assert await c_nodes[0].set("KEY17", "value")
+    assert await c_nodes[1].set("KEY18", "value")
+    assert await c_nodes[1].set("KEY19", "value")
+    assert await c_nodes[0].execute_command("DBSIZE") == 10
+
+    res = await c_nodes_admin[1].execute_command(
+        "DFLYCLUSTER", "START-SLOT-MIGRATION", "127.0.0.1", str(nodes[0].admin_port), "3000", "9000"
+    )
+    assert "OK" == res
+
+    await asyncio.sleep(1)
+
+    status = await c_nodes_admin[1].execute_command(
+        "DFLYCLUSTER", "SLOT-MIGRATION-STATUS", "127.0.0.1", str(nodes[0].admin_port)
+    )
+    assert "STABLE_SYNC" == status
```

Comment:
Can we issue `SLOT-MIGRATION-STATUS` in a loop, without sleeping before, and with a short sleep in between? And finish the loop when we reach stable sync
That will make the test both faster and more stable

Reply:
I think we can